File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
import java.math.*;
class Solution {
    public int solution(int balls, int share) {
        int answer = 0;
        
        BigInteger operation;
        
        operation = factorial(balls).divide(factorial(balls - share).multiply(factorial(share)));
        
        answer = operation.intValue();
        
        return answer;
    }
    
    public BigInteger factorial(int n){
        BigInteger temp = BigInteger.valueOf(1);
        
        for(int i = 1; i <= n; i++){
            temp = temp.multiply(BigInteger.valueOf(i));
        }
        
        return temp;
    }
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long oper(int a, int b){
    int n = (a - b) > b ? b : a - b;

    if(n == 0) return 1;

    return oper(a - 1, n - 1) * a / n;
}

int solution(int balls, int share) {
    int answer = (int) oper(balls, share);
    
    return answer;
}
